apps/server/src/modules/user-login-migration/service/user-login-migration-rollback.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@nestjs/common';
+import { EntityId, UserLoginMigrationDO } from '@shared/domain';
+import { LegacyLogger } from '@src/core/logger';
+import { SchoolMigrationService } from './school-migration.service';
+import { UserLoginMigrationService } from './user-login-migration.service';
+import { RollbackUserLoginMigrationError } from '../error/rollback-user-login-migration.error';
+
+@Injectable()
+export class UserLoginMigrationRollbackService {
+	constructor(
+		private readonly schoolMigrationService: SchoolMigrationService,
+		private readonly userLoginMigrationService: UserLoginMigrationService,
+		private readonly logger: LegacyLogger
+	) {}
+
+	// TODO: 839 rename this service to an uc and fix usages.
+	async rollbackIfNecessary(schoolId: EntityId): Promise<void> {
+		const hasSchoolMigratedUser = await this.schoolMigrationService.hasSchoolMigratedUser(schoolId);
```

Comment:
My first thought was: "The easier way would have been to check if the school has migrated by checking when it was last migrated", but then I saw that we don't even have an attribute for that. Maybe we should add that.
Then we could add an OR case here, in case something goes wrong during user migration and the school gets migrated, but the user does not. Then we try to cancel the migration, but the school has already migrated, which is a condition to not cancel anymore.